File: 2019/day02/python/part1.py
#!/usr/bin/env python

import sys

OPADD = 1
OPMULTIPLY = 2
OPHALT = 99 

num = [int(i) for i in open(sys.argv[1], "r").read().split(',')]

num[1] = 12
num[2] = 2

i = 0

while True:
    one = num[i + 1]
    two = num[i + 2]
    three = num[i + 3]
    if num[i] == OPADD:
        num[three] = num[one] + num[two]
        i += 4
    elif num[i] == OPMULTIPLY:
        num[three] = num[one] * num[two]
        i += 4
    elif num[i] == OPHALT:
        break
    else:
        print("?")
        
print("Solution: " + str(num[0]))

File: 2019/day02/lua/part1.lua
#!/usr/bin/env luajit

local OPADD = 1
local OPMULTIPLY = 2
local OPHALT = 99

local num = {}
for str in io.lines(arg[1])():gmatch("%d+") do
    num[#num+1] = tonumber(str)
end

num[2] = 12
num[3] = 2

local i = 1

while true do
    one = num[i + 1]+1
    two = num[i + 2]+1
    three = num[i + 3]+1
    if num[i] == OPADD then
        num[three] = num[one] + num[two]
        i = i + 4
    elseif num[i] == OPMULTIPLY then
        num[three] = num[one] * num[two]
        i = i + 4
    elseif num[i] == OPHALT then
        break
    end
end

print("Solution: " .. num[1])